File: kotlin-back/src/main/kotlin/com/example/kotlinback/member/service/MemberService.java
package com.example.kotlinback.member.service;

import com.example.kotlinback.member.dto.LoginDto;
import com.example.kotlinback.member.dto.MemberDto;
import com.example.kotlinback.member.dto.SignUpDto;

import java.util.Map;

public interface MemberService {

    void signUp(SignUpDto signUpDto);

    Map<String, String> login(LoginDto loginDto);

    void confirmMemberByNickname(String nickname);

    void confirmMemberByUsername(String username);

    MemberDto getByUsername(String username);

    String getAccessTokenWithRefreshToken(String refreshToken);

    void signOut(String memId);

    void sendEmailAndSaveTempData(String email);

    void confirmEmailAndCode(String email, String code);
}

File: kotlin-back/src/main/kotlin/com/example/kotlinback/member/service/MemberService.kt
package com.example.kotlinback.member.service

import com.example.kotlinback.member.dto.LoginDto
import com.example.kotlinback.member.dto.MemberDto
import com.example.kotlinback.member.dto.SignUpDto

interface MemberService {
    fun signUp(signUpDto: SignUpDto)
    fun login(loginDto: LoginDto): Map<String, String>
    fun confirmMemberByNickname(nickname: String)
    fun confirmMemberByUsername(username: String)
    fun getByUsername(username: String): MemberDto
    fun getAccessTokenWithRefreshToken(refreshToken: String): String
    fun signOut(memId: String)
    fun sendEmailAndSaveTempData(email: String)
    fun confirmEmailAndCode(email: String, code: String)
}
